### PR TITLE
[test] Add multi memory replicas test cases

### PR DESCRIPTION
**Pull request by ThreadDao**
- Add multi memory replicas test cases
Issues: #16648 

Signed-off-by: ThreadDao <yufen.zong@zilliz.com>

### CODE REVIEW

File: tests/python_client/testcases/test_collection.py [hunk 1]
```diff
@@ -2161,6 +2162,254 @@ def test_load_partitions_release_collection(self):
...
+        method: load with invalid replica number
+        expected: raise exception
+        """
+        # create, insert
+        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
+        df = cf.gen_default_dataframe_data()
+        insert_res, _ = collection_w.insert(df)
+        assert collection_w.num_entities == ct.default_nb
+
+        collection_w.load(replica_number=replicas)
```
Comment (yanliang567) [inline tests/python_client/testcases/test_collection.py:2204]: no errors here for invalid replicas?

File: tests/python_client/testcases/test_collection.py [hunk 2]
```diff
@@ -2161,6 +2162,254 @@ def test_load_partitions_release_collection(self):
...
+        """
+        # create, insert
+        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
+        df = cf.gen_default_dataframe_data()
+        insert_res, _ = collection_w.insert(df)
+        assert collection_w.num_entities == ct.default_nb
+
+        collection_w.load(replica_number=1)
+        collection_w.query(expr=f"{ct.default_int64_field_name} in [0]")
+        collection_w.load(replica_number=2)
```
Comment (yanliang567) [inline tests/python_client/testcases/test_collection.py:2245]: do we expect an error or warning here when loading with a different replicas

File: tests/python_client/testcases/test_collection.py (hunk 1 above)
Comment (ThreadDao) [reply tests/python_client/testcases/test_collection.py:2204]: No exception is thrown by design. This is for compatibility with older versions without the replica_number field, and the value obtained from the request is 0. So if the replica_number is None or less than 0, milvus will set it to 1.

File: tests/python_client/testcases/test_collection.py (hunk 2 above)
Comment (ThreadDao) [reply tests/python_client/testcases/test_collection.py:2245]: I agree with throwing exceptions. However, there is a problem with the case itself. We can go ahead and modify it later

**Review by yanliang567 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**